app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -603,14 +603,11 @@ def repos_pending_sync(environment = nil, content_view = nil)
       end
 
       def rhsm_url
-        # Since Foreman 3.1 this setting is set
-        if (rhsm_url = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url'))
+        # Since Foreman 3.1 this setting is set. No value means direct use
+        if (rhsm_url_setting = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url').presence)
+          URI(rhsm_url_setting)
+        else
           URI(rhsm_url)
```

Comment:
Isn't that recursive now?

Reply:
I think it is. I hate Rails' URL helpers

Reply:
Sure, but where should `rhsm_url` come from even w/o Rails?
I think the following should work?
```ruby
def rhsm_url
  if (rhsm_url_setting = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url').presence)
    URI(rhsm_url_setting)
  else
    URI("https://#{URI.parse(url).host}/rhsm")
  end
end
```

Reply:
Rails itself should be able to look up the route and generate the url. The whole change here is also that it's the Foreman URL now, not on the proxy. Essentially making it opt in whether there is a reverse proxy for RHSM

Reply:
I ended up using the `foreman_url` setting and hardcode the path instead of figuring out the URL helpers. They don't appear to be included in the helper and passing in context is going to be too hard (and possibly wrong).